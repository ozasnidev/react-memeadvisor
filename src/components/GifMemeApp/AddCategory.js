import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

    const [newCategory, setNewCategory] = useState('');
    const handleNewInputValue = (event) => {
        setNewCategory(event.target.value);
    }
    const handleSubmit = (event) => {
        event.preventDefault();
        if(newCategory.trim().length > 2){
            setCategories(categories => [newCategory, ...categories]);
            setNewCategory('');
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <div className="panel-block">
            <p className="control">
                <input className="input is-success" 
                    type="text" 
                    value={newCategory}
                    onChange={handleNewInputValue}
                    placeholder="Search" 
                />
            </p>
            </div>
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
