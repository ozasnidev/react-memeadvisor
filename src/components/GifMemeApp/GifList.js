import React from 'react'
import { useFetchGif } from '../hooks/useFetchGif';
import PropTypes from 'prop-types';

export const GifList = ({category}) => {

    const {data,loading} = useFetchGif( category );

    return (
        <div>
            <div className="panel-block">
                <h4 className="title is-4 animate__animated animate__fadeInDown"> {category} </h4>
                {loading && <div className="spinner"></div>}
            </div>
            <div className="img-thumbnail-container">
                {data.map(image => ( 
                    <img src={image.url} alt={image.title} key={image.id} className="img-thumbnail  animate__animated animate__fadeInUp"/>
                ))}
            </div>
        </div>
    )
}

GifList.propTypes = {
    category: PropTypes.string.isRequired
}