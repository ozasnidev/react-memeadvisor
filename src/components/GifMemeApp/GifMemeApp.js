import React, {useState} from 'react';
import NavbarMeme from '../Shared/navbar/Navbar';
import { AddCategory } from './AddCategory';
import { GifList } from './GifList';


const GifMemeApp = ({ defaultCategories = []}) => { 
    const [categories, setCategories] = useState(defaultCategories);

    return (
        <>
            <NavbarMeme />
            <div className="container">
                <div className="add-category-container">
                    <article className="panel is-success">
                        <p className="nav-title panel-heading">Categories</p>
                        <AddCategory setCategories = {setCategories}/>
                        {categories.map(category => <GifList key={category} category={category} />)} 
                    </article>
                </div>
            </div>
        </>
    )   
}
export default GifMemeApp;