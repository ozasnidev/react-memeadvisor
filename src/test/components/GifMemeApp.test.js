import { shallow } from 'enzyme';
import React from 'react';
import GifMemeApp from '../../components/GifMemeApp/GifMemeApp';

describe('GifMemeApo Component test', () => {

    test('should show the component without issues', () => {
        const wrapper = shallow(<GifMemeApp />);
        expect(wrapper).toMatchSnapshot();
    });
    test('should show a category list', () => {
        const categoryMock = ['Donald Trump', 'Joe Biden'];
        const wrapper = shallow(<GifMemeApp defaultCategories = {categoryMock} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifList').length).toBe(categoryMock.length);
    });
    
});