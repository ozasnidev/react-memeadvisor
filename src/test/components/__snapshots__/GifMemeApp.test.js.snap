// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifMemeApo Component test should show a category list 1`] = `
<Fragment>
  <NavbarMeme />
  <div
    className="container"
  >
    <div
      className="add-category-container"
    >
      <article
        className="panel is-success"
      >
        <p
          className="nav-title panel-heading"
        >
          Categories
        </p>
        <AddCategory
          setCategories={[Function]}
        />
        <GifList
          category="Donald Trump"
          key="Donald Trump"
        />
        <GifList
          category="Joe Biden"
          key="Joe Biden"
        />
      </article>
    </div>
  </div>
</Fragment>
`;

exports[`GifMemeApo Component test should show the component without issues 1`] = `
<Fragment>
  <NavbarMeme />
  <div
    className="container"
  >
    <div
      className="add-category-container"
    >
      <article
        className="panel is-success"
      >
        <p
          className="nav-title panel-heading"
        >
          Categories
        </p>
        <AddCategory
          setCategories={[Function]}
        />
      </article>
    </div>
  </div>
</Fragment>
`;
